src/services/goToDefinition.ts

```diff
@@ -510,4 +522,26 @@ namespace ts.GoToDefinition {
                 return false;
         }
     }
+
+    interface StatementWithExpression extends Statement {
+        readonly expression: Expression
+    }
+
+    function createDefinitionInfoFromStatement<K extends KeywordSyntaxKind>(statement: StatementWithExpression, keywordKind: K, name: string): DefinitionInfo {
+        const sourceFile = statement.getSourceFile();
+        const [keyword] = statement.getChildren().filter((node): node is Token<K> => node.kind === keywordKind);

```

Comment:
We have a `find` helper you can use instead of `filter`.

```suggestion
        const keyword = find(statement.getChildren(sourceFile), (node): node is Token<K> => node.kind === keywordKind)!;
```

As an aside, it's a little "risky" always assuming the `switch` keyword is around - the language service has a tendency to find places where the parser generates nodes that we would never have expected; however, I think it should be okay in this case since I cannot imagine how else we would end up with a switch statement otherwise.

Reply:
fixed